src/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -741,7 +741,15 @@ public override bool Equals(object comparand)
 
         public override int GetHashCode()
         {
-            return (Name + "=" + Value + ";" + Path + "; " + Domain + "; " + Version).GetHashCode();
+            // Copied from Tuple.CombineHashCodes
+            int hash1 = Name.GetHashCode();
+            hash1 = ((hash1 << 5) + hash1) ^ Value.GetHashCode();
```

Comment:
I know this has a lot of bitshifts and stuff but I'm not convinced that (33*hash1)^hash2 wouldn't be faster (`<<5` is equivalent to multiplying by 32)